include/mc_rtc/gui/Form.h

```diff
@@ -189,28 +189,43 @@ struct FormComboInput : public FormElement<FormComboInput, Elements::ComboInput>
   inline FormComboInput(const std::string & name,
                         bool required,
                         const std::vector<std::string> & values,
+                        int def,
                         bool send_index = false)
-  : FormElement<FormComboInput, Elements::ComboInput>(name, required), values_(values), send_index_(send_index)
+  : FormElement<FormComboInput, Elements::ComboInput>(name, required), values_(values), def_(def),
+    send_index_(send_index), has_def_(true)
   {
   }
 
+  inline FormComboInput(const std::string & name,
+                        bool required,
+                        const std::vector<std::string> & values,
+                        bool send_index = false)
+  : FormComboInput(name, required, values, -1, send_index)
+  {
+    has_def_ = false;
+  }
+
   static constexpr size_t write_size_()
   {
-    return 2;
+    return 4;
   }
 
   inline void write_(mc_rtc::MessagePackBuilder & builder)
   {
     builder.write(values_);
+    builder.write(def_);
     builder.write(send_index_);
+    builder.write(has_def_);
   }
 
   /** Invalid element */
   inline FormComboInput() {}
 
 private:
   std::vector<std::string> values_;
+  int def_;
   bool send_index_ = false;
+  bool has_def_;
```

Comment:
I think the client can treat `-1` or any negative value for `def` as meaning "no default" so we don't need the `has_def_` field